From cf93285d57faa5a0167bd5d6aa7ab1bbeabf4937 Mon Sep 17 00:00:00 2001
From: Guenter Roeck <groeck@chromium.org>
Date: Mon, 6 Apr 2020 18:27:27 -0700
Subject: [PATCH 23/61] FIXUP: BACKPORT: FROMLIST: virtio_video: Add the Virtio
 Video V4L2 driver

0-day reports:

drivers/media/virtio/virtio_video_driver.c:103:5-11: ERROR:
	allocation function on line 102 returns NULL not ERR_PTR on failure

The various basic memory allocation functions don't return ERR_PTR.

BUG=b:120456557, b:151703605
TEST=compile with VIRTIO_VIDEO enabled

Change-Id: Ia556120dd001a4f395bcfb66a29f5f3c81ee8dd0
Fixes: bbca48582a73 ("BACKPORT: FROMLIST: virtio_video: Add the Virtio Video V4L2 driver")
Reported-by: kbuild test robot <lkp@intel.com>
Signed-off-by: Guenter Roeck <groeck@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/2138634
Reviewed-by: Alexandre Courbot <acourbot@chromium.org>
Reviewed-by: Keiichi Watanabe <keiichiw@chromium.org>
---
 drivers/media/virtio/virtio_video_driver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/virtio/virtio_video_driver.c b/drivers/media/virtio/virtio_video_driver.c
index f01b744587c8..6116c3a53d11 100644
--- a/drivers/media/virtio/virtio_video_driver.c
+++ b/drivers/media/virtio/virtio_video_driver.c
@@ -100,8 +100,8 @@ static int virtio_video_query_cap_resp_buf(struct virtio_video *vv, void
 	int resp_size = vv->max_caps_len;
 
 	*resp_buf = kzalloc(vv->max_caps_len, GFP_KERNEL);
-	if (IS_ERR(*resp_buf)) {
-		ret = PTR_ERR(*resp_buf);
+	if (!*resp_buf) {
+		ret = -ENOMEM;
 		goto err;
 	}
 
-- 
2.25.1

